src/AzureClient/AzureEnvironment.cs

```diff
@@ -75,7 +75,17 @@ public static AzureEnvironment Create(string subscriptionId)
 
             // Register the token cache for serialization
             var cacheFileName = "iqsharp.bin";
-            var storageCreationProperties = new StorageCreationPropertiesBuilder(cacheFileName, cacheDirectory, ClientId).Build();
+            var storageCreationProperties = new StorageCreationPropertiesBuilder(cacheFileName, cacheDirectory, ClientId)
+                .WithMacKeyChain(
+                    serviceName: "Microsoft.Quantum.IQSharp",
+                    accountName: "MSALCache")
```

Comment:
Please verify on Mac and Linux before completing.

Reply:
Yes, already done, thanks for confirming.